app/models/base.py

```diff
@@ -82,40 +78,98 @@ def related_fields(self) -> List[str]:
         and not Case.person.income
-
         Todo: Allow deeper level of nested fields to be translated
+
+        Args:
+            session: active database session
+            create: whether to create a new instance of the model. This is particularly useful for when fetching
+                    related instances if the data in the request contains the id field
         """
-        data = self.model_dump()
+        data = self.model_dump(exclude_unset=True)
         related_fields_names = self.related_fields
         related_fields_data = {}
         fields = {}
         for field_name, field_value in data.items():
-            if field_name in related_fields_names:
+            if field_name in related_fields_names and field_value:
                 related_fields_data[field_name] = field_value
             else:
                 fields[field_name] = field_value
-        return {**fields, **self._translate_related_fields(related_fields_data)}
-
-    def _translate_related_fields(self, fields) -> dict:
+        return {
+            **fields,
+            **self._translate_related_fields(session, related_fields_data, create),
+        }
+
+    def _translate_related_fields(
+        self, session: Session, fields: dict, create: bool = False
+    ) -> dict:
         """Convert related fields from a dict to an instance of their declared model"""
         instances = {}
         for field_name, field_value in fields.items():
             field = getattr(self, field_name)
             if not field:
                 continue
             if isinstance(field, list):
-                model = field[0].model
+                request_class = field[0]
             else:
-                model = field.model
+                request_class = field
 
             if isinstance(field_value, list):
-                instances[field_name] = [model(**value) for value in field_value]
+                instances[field_name] = []
+                for value in field_value:
+                    instance = request_class.dict_to_instance(
+                        session, request_class.model, value, create
+                    )
+                    instances[field_name].append(instance)
             else:
-                instances[field_name] = model(**field_value)
+                instance = request_class.dict_to_instance(
+                    session, request_class.model, field_value, create
+                )
+                instances[field_name] = instance
         return instances
+
+    def dict_to_instance(
+        self, session: Session, model: SQLModel, values: dict, create: bool = False
+    ) -> SQLModel:
+        return model(**values)
+
+
+class BaseUpdateRequest(BaseRequest):
+    id: uuid.UUID | None = None
+
+    def dict_to_instance(
+        self, session: Session, model: SQLModel, values: dict, create: bool = False
+    ) -> SQLModel:
+        if not create and "id" in values:
+            instance = session.get(model, values["id"])
```

Comment:
If this query returns `None` we should propagate this error back up rather than trying to call `sqlmodel_update(values)` on `None`.